tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/amqp/connect/PagedMirrorTest.java

```diff
@@ -0,0 +1,161 @@
+ */
+
+package org.apache.activemq.artemis.tests.integration.amqp.connect;
+
+import javax.jms.Connection;
+import javax.jms.ConnectionFactory;
+import javax.jms.MessageConsumer;
+import javax.jms.MessageProducer;
+import javax.jms.Queue;
+import javax.jms.Session;
+import javax.jms.TextMessage;
+import java.io.File;
+import java.util.HashMap;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import org.apache.activemq.artemis.core.config.amqpBrokerConnectivity.AMQPBrokerConnectConfiguration;
+import org.apache.activemq.artemis.core.config.amqpBrokerConnectivity.AMQPMirrorBrokerConnectionElement;
+import org.apache.activemq.artemis.core.persistence.impl.journal.JournalRecordIds;
+import org.apache.activemq.artemis.core.server.ActiveMQServer;
+import org.apache.activemq.artemis.tests.util.ActiveMQTestBase;
+import org.apache.activemq.artemis.tests.util.CFUtil;
+import org.apache.activemq.artemis.tests.util.Wait;
+import org.jboss.logging.Logger;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+public class PagedMirrorTest extends ActiveMQTestBase {
+
+   private static final Logger logger = Logger.getLogger(PagedMirrorTest.class);
+   ActiveMQServer server1;
+
+   ActiveMQServer server2;
+
+   @Before
+   @Override
+   public void setUp() throws Exception {
+      super.setUp();
+
+      server1 = createServer(true, createDefaultConfig(0, true), 1024, 10 * 1024);
+      server1.getConfiguration().getAcceptorConfigurations().clear();
+      server1.getConfiguration().addAcceptorConfiguration("server", "tcp://localhost:61616");
+      AMQPBrokerConnectConfiguration brokerConnectConfiguration = new AMQPBrokerConnectConfiguration("other", "tcp://localhost:61617").setReconnectAttempts(-1).setRetryInterval(1000);
+      brokerConnectConfiguration.addElement(new AMQPMirrorBrokerConnectionElement());
+      server1.getConfiguration().addAMQPConnection(brokerConnectConfiguration);
+
+      server2 = createServer(true, createDefaultConfig(1, true), 1024, 10 * 1024);
+      server2.getConfiguration().getAcceptorConfigurations().clear();
+      server2.getConfiguration().addAcceptorConfiguration("server", "tcp://localhost:61617");
+      brokerConnectConfiguration = new AMQPBrokerConnectConfiguration("other", "tcp://localhost:61616").setReconnectAttempts(-1).setRetryInterval(1000);
+      brokerConnectConfiguration.addElement(new AMQPMirrorBrokerConnectionElement());
+      server2.getConfiguration().addAMQPConnection(brokerConnectConfiguration);
+
+      server1.start();
+      server2.start();
+   }
+
+   @Test
+   public void testPaged() throws Throwable {
+      String sendURI = "tcp://localhost:61616";
+      String consumeURI = "tcp://localhost:61616";
+      String secondConsumeURI = "tcp://localhost:61617";
+
+      Wait.waitFor(() -> server1.locateQueue("$ACTIVEMQ_ARTEMIS_MIRROR_other") != null);
+
+      org.apache.activemq.artemis.core.server.Queue snf1 = server2.locateQueue("$ACTIVEMQ_ARTEMIS_MIRROR_other");
```

Comment:
May have been clearer not to name both connections "other". 

Also seems odd to wait for the server1 queue to appear and then locate the server2 queue first. Seems like you would wait on and locate each in turn.